### PR TITLE
AutoRewind

### PR DESCRIPTION
**Pull request by axoTheLotl**
Here is the autorewind functionality. 
Defaults to zero as discussed.
Autorewind is added to play() method of the MediaPlayerService so it's "rewinding" even if file starts from beginning. Since seekTo() jumps to zero if msc is negative this doesn't appear to be a problem. 

Let me know if any changes need to be made.
best
Lotl

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,164 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (flackbash) [inline .idea/codeStyles/Project.xml]: This file should not be commited. Only add those files to the commit in which you actively made changes that are part of the implementation of this feature, i.e. MediaPlayerService.java, settings.xml and strings.xml.

File: app/src/main/java/com/prangesoftwaresolutions/audioanchor/MediaPlayerService.java [hunk 2]
```diff
@@ -436,7 +438,6 @@ public void onStop() {
                 //Stop the service
                 stopSelf();
             }
-
```
Comment (flackbash) [inline app/src/main/java/com/prangesoftwaresolutions/audioanchor/MediaPlayerService.java]: Keep the newline here. In general, try not to make any changes - even small ones like adding a newline or an unneeded import - if they are not part of the feature you are implementing and do not improve the code.

File: app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java [hunk 3]
```diff
@@ -128,6 +128,7 @@ protected void onCreate(Bundle savedInstanceState) {
         mLastSleepTime = sharedPreferences.getInt(getString(R.string.preference_last_sleep_key), Integer.valueOf(getString(R.string.preference_last_sleep_val)));
         mDirectory = sharedPreferences.getString(getString(R.string.preference_filename), null);
 
+
```
Comment (flackbash) [inline app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java]: Same as mentioned in the comment for the MediaPlayerService. Don't include this file in your commit.

File: app/src/main/res/xml/settings.xml [hunk 4]
```diff
@@ -24,6 +24,14 @@
         android:defaultValue="@string/settings_dark_default"
         android:key="@string/settings_dark_key"
         android:title="@string/settings_dark_label"/>
+    <EditTextPreference
+        android:defaultValue="0"
```
Comment (flackbash) [inline app/src/main/res/xml/settings.xml]: Here you should use a reference to the default value you specified in the strings.xml file.

File: app/src/main/res/xml/settings.xml [hunk 5]
```diff
@@ -24,6 +24,14 @@
...
+        android:inputType="number"
+        android:key="@string/settings_autorewind_key"
+        android:selectAllOnFocus="true"
+        android:singleLine="true"
+        android:maxLength="1"
```
Comment (flackbash) [inline app/src/main/res/xml/settings.xml]: Better set maxLength to 2. I think it is plausible that some people would like to set the autorewind time to more than 9 seconds.

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (Sulter) [reply .idea/codeStyles/Project.xml]: I would suggest adding .idea to .gitignore. This keeps happening.

Comment (Sulter) [reply .idea/codeStyles/Project.xml]: I would suggest adding .idea to .gitignore? This seems to keep happning.

**Comment by axoTheLotl** [comment]
Thanks a lot for your comments!

I hope everything is correct now. Still having a difficult time with git but I think it should be fine.

**Comment by flackbash** [comment]
Perfect :)

**Pull request merged**